src/lib/ota/common.ts

```diff
@@ -313,40 +327,53 @@ function callOnProgress(startTime: number, lastUpdate: number, imageBlockRequest
         percentage = Math.round(percentage * 10000) / 100;
-        logger.debug(`OTA update at ${percentage}%, remaining ${remaining} seconds`);
+        logger.debug(`OTA: Update at ${percentage}%, remaining ${remaining} seconds`);
         onProgress(percentage, remaining === Infinity ? null : remaining);
         return now;
     } else {
         return lastUpdate;
     }
 }
 
-export async function isUpdateAvailable(device: Zh.Device, logger: Logger, isNewImageAvailable: IsNewImageAvailable, requestPayload: Ota.ImageInfo,
-    getImageMeta: Ota.GetImageMeta = null) {
-    logger.debug(`Check if update available for '${device.ieeeAddr}' (${device.modelID})`);
+export async function isUpdateAvailable(device: Zh.Device, logger: Logger, requestPayload: Ota.ImageInfo,
+    isNewImageAvailable: IsNewImageAvailable = null, getImageMeta: Ota.GetImageMeta = null) {
+    logger.debug(`OTA: Checking if update available for '${device.ieeeAddr}' (${device.modelID})`);
 
     if (requestPayload === null) {
         const endpoint = getOTAEndpoint(device);
         assert(endpoint != null, `Failed to find endpoint which support OTA cluster`);
-        logger.debug(`Using endpoint '${endpoint.ID}'`);
+        logger.debug(`OTA: Using endpoint '${endpoint.ID}'`);
 
         const request = await requestOTA(endpoint);
-        logger.debug(`Got OTA request '${JSON.stringify(request.payload)}'`);
+        logger.debug(`OTA: Got request '${JSON.stringify(request.payload)}'`);
         requestPayload = request.payload;
     }
 
     const availableResult = await isNewImageAvailable(requestPayload, logger, device, getImageMeta);
-    logger.debug(`Update available for '${device.ieeeAddr}': ${availableResult.available < 0 ? 'YES' : 'NO'}`);
+    logger.debug(`OTA: Update available for '${device.ieeeAddr}': ${availableResult.available < 0 ? 'YES' : 'NO'}`);
     if (availableResult.available > 0) {
-        logger.warn(`Firmware on '${device.ieeeAddr}' is newer than latest firmware online.`);
+        logger.warn(`OTA: Firmware on '${device.ieeeAddr}' is newer than latest firmware online.`);
     }
     return {...availableResult, available: availableResult.available < 0};
 }
 
 export async function isNewImageAvailable(current: Ota.ImageInfo, logger: Logger, device: Zh.Device, getImageMeta: Ota.GetImageMeta) {
+    const currentS = JSON.stringify(current);
+    logger.debug(`OTA: Is new image available for '${device.ieeeAddr}', current '${currentS}'`);
     const meta = await getImageMeta(current, logger, device);
-    const [currentS, metaS] = [JSON.stringify(current), JSON.stringify(meta)];
-    logger.debug(`Is new image available for '${device.ieeeAddr}', current '${currentS}', latest meta '${metaS}'`);
+
+    // Soft-fail because no images in repo/URL for specified device
+    if (!meta) {
+        logger.warn(`OTA: Images currently unavailable for device '${device.modelID}', hardwareVersion '${device.hardwareVersion}', ${currentS}'`);
```

Comment:
Please also log the `device.manufacturerName` here (it's relevant for TuYa devices)